common-content/en/module/legacy/adding/index.md

```diff
@@ -0,0 +1,144 @@
+Then I should no longer be following sample  
+And the unfollow button is not visible
+
+The tabs contain sample code for each step of this process, but you should write your own implementation, based on your understanding of the Purple Forest codebase.
+
+Before you start coding, open each file you think you will need in your editor. You will need to touch 5-7 files only. Use what you understand about the system to predict which files these will be.
+
+===[[Step 2: Write a test]]===
+
+```javascript
+test("allows unfollowing a user from their profile", async ({ page }) => {
+  // Given a profile component sample
+  // And I am logged in as sample2
+  await loginAsSample2(page);
+  // And sample2 is following sample
+  await page.goto("/front-end/#/profile/sample");
+  await page.click('[data-action="follow"]');
+
+  // When I view the profile component for sample
+  // Then I should see a button labeled "Unfollow"
+  const unfollowButton = page.locator('[data-action="unfollow"]');
+  await expect(unfollowButton).toBeVisible();
+
+  // When I click the "Unfollow" button
+  await unfollowButton.click();
+
+  // Then I should no longer be following sample
+  const followerCount = page.locator("[data-follower-count]");
+  await expect(followerCount).toHaveText("0");
+  // And the unfollow button is not visible
+  await expect(unfollowButton).toBe("hidden");
+});
+```
+
+Commit your changes to your branch.
+
+===[[Step 3: Implement the UI]]===
+
+#### 🏠 `index.html`
+
+Find the follow button and, and following its patterns, add an unfollow button next to it.
+
+#### 🪪 `components/profile.mjs`
+
+`handleUnfollow`: Add a new function to handle the unfollow action. Find the follow handler and use it as a template.
+
+`createProfile` : For each line that creates the follow button, add a line that creates the unfollow button. For example, after:
+
+`const followButtonEl = profileElement.querySelector("[data-action='follow']");`
+
+add
+
+`const unfollowButtonEl = profileElement.querySelector("[data-action='unfollow']");`
+
+Don't feel tempted to optimise this yet. You can refactor later.
+
+### 🍱 `views/profile.mjs`
+
+You should have exported the `handleUnfollow` function from the profile component. Now you need to import it into the profile view and call it when the unfollow button is clicked.
+
+Find the follow button event listener and add a similar listener for the unfollow button.
+
+Commit your changes to your branch.
+
+Your test is still failing. Make a prediction about what will happen when you click the unfollow button. What will you need to change to make the test pass?
+
+Use your debugging skills to find out.
+
+===[[Step 4: Hook up the back end]]===
+
+(You could have done this from the back to the front, we just happened to start at the front end.)
+
+You have written an interface, but you haven't connected it to anything in the back end! There's an API endpoint in the apiService, but is there an matching endpoint in `main.py`? Go look.
```

Comment:
Are we expecting to have a dangling `unfollow` in `apiService`? I probably wouldn't?

Reply:
I have left one in, purely for this reason -- at the moment there's almost no mention of the backend and it's getting long. Depending on what you build, we will need to do some exploration of the back end, so I need some space to do that.

Totally open to revise this, in fact intend to. (But we do need some space for the back end, so some part will have to be cut down)

Reply:
Note here: the plan being that if we take out the apiService unfollow, we would

1. In your map of this system, you should already have identified that the front end fetches data from the backend in the apiService. Go to this file, find the follow function... etc (basically exactly as we have already written, but doing it first in apiService and then in main.py)